### DISCUSSION TITLE
Subproject 'shared-mime-info' did not override 'shared-mime-info' dependency and no variable name specified

### DISCUSSION THREAD

**Discussion by alanocallaghan** [mesonbuild/meson#11799] [Q&A] [answered]
I am trying to cross-build [openslide](https://github.com/alanocallaghan/openslide-winbuild/) for linux, and gdk-pixbuf requires shared-mime-info. When I add it as a dependency for my meson project (rather than just implicitly via gdk-pixbuf) and include a wrap file as follows:

```
[wrap-file]
directory = shared-mime-info-2.2
source_url = https://gitlab.freedesktop.org/xdg/shared-mime-info/-/archive/2.2/shared-mime-info-2.2.tar.gz
source_filename = shared-mime-info-2.2.tar.gz
source_hash = bcf5d552318136cf7b3ae259975f414fbcdc9ebce000c87cf1f0901ff14e619f

[provide]
dependency_names = shared-mime-info
```

I get the error from the title. I have seen [this discussion](https://github.com/mesonbuild/meson/discussions/10484), but it doesn't seem applicable here.

How can I tell my wrap file to override the dependency?

**Comment by eli-schwartz** [comment]
That discussion is applicable here (although there are extra nuances for shared-mime-info). There are two ways to specify a dependency in a wrap file:
- using `dependency_names` which requires that the subproject collaborate by using the `meson.override_dependency()` method in its meson.build files
- using `shared-mime-info = shared_mime_info_dep` where the left hand side is the name of the dependency, and the right hand side is a meson variable that was assigned like this: `shared_mime_info_dep = declare_dependency(...)`

**Reply by alanocallaghan** [reply]
So if the project uses neither `override_dependency` nor `declare_dependency`, I'm out of luck?

**Reply by eli-schwartz** [reply, marked as answer]
You can submit a pull request to the project, or you can create a `*.patch` file, then use it with `diff_files` as described here: https://mesonbuild.com/Wrap-dependency-system-manual.html#diff-files

Warning: requires additional build dependencies, `patch` or `git apply` may not be installed on all build machines. Also, it requires relatively recent versions of meson itself.

**Reply by alanocallaghan** [reply]
Thanks! Sorry if it's a simple question, I've been doing my best to get through the docs but am jumping around a bit as I encounter issues. Very much appreciate your work and the speed of answering this is phenomal

Cheers

**Reply by eli-schwartz** [reply]
No problem, that's what we're here for. :)

(After all, this is the discussion and Q&A section, not the bug reports section. :D)

Also this (diff_files) is relatively new so there's less hive mind knowledge about it.
